AGW-4438: Spike Control policy revision (#298)
mulesoft/docs-api-manager: modules/ROOT/pages/spike-control-reference.adoc

| Method & Resource conditions | The option to add configurations to only a select few or all methods and resources of the API | Yes
|===

To enforce request quotas, see the xref:rate-limiting-sla-based-policy.adoc[Rate-Limiting SLA policy] and xref:rate-limiting.adoc[Rate Limiting policy].

== FAQ

*When does the sliding window start?*

The window starts with the first request after the policy is successfully applied.

*What type of window does the algorithm use?*

> The algorithm uses a sliding window by monitoring the transactions for the amount of time for which you configure in the policy.

Reviewer: Shouldn't this be the first entry rather than the second?

Reviewer: ```suggestion
*What type of window does the algorithm use?*

The algorithm uses a sliding window that monitors transactions for the amount of time that you configure in the policy.
```

Reviewer 2: Done! Thanks!

*What happens when the quota is exhausted?*

> When the request quota is exhausted, the Spike Control policy queues the request and retries as configured. The request is rejected if no quota is remaining after several retries. Quota for other requests is only available after the first request has elapsed the time in millisecconds as configured in the policy. Requests are processed in cycles using the configured time limit in this way.

Reviewer: ```suggestion
When the request quota is exhausted, the Spike Control policy queues the request and retries as configured. The request is rejected if no quota is remaining after several retries. Quota for other requests is available only after the first request exceeds the time limit configured in the policy; in this way, requests are processed in cycles.
```

Reviewer 2: Done! Thanks!

*Can I configure multiple windows?*

> No. The Spike Control policy ensures that the backend server does not serve more requests than it can handle. If you need accountability, use either Rate-Limiting SLA or Rate Limit policies instead.

Reviewer: This implies that the backend can manage no more than one window; did you intend that?

Reviewer 2: Yes, Thanks! Done!

[[response-header]]
*What does each response header mean?*

Each response header has information about the current state of the request:

> * X-Ratelimit-Remaining: The amount of available quota

Reviewer: "currently available"? (because of "remaining")

Reviewer 2: Done! Thanks!

* X-Ratelimit-Limit: The maximum available requests per window

> * X-Ratelimit-Reset: The remaining time, in milliseconds, until the oldest request ages enough to fall outside of the sliding window. If quota is still available, then this header is set to 0 (zero), as the algorithm can still serve new requests.

Reviewer: "as" > "because," if that is what you really mean

Reviewer: Also, bullet-list format:

```suggestion
* X-Ratelimit-Reset: The remaining time, in milliseconds, until the oldest request ages enough to fall outside of the sliding window. 
+
If quota is still available, then this header is set to 0 (zero), as the algorithm can still serve new requests.
```

Reviewer 2: Done! Thanks!

> MuleSoft advises that you expose these headers only if the API is used within the organization. In a public API scenario, this data can be used maliciously to invoke spikes in traffic that leaves your API without any quota left.

Reviewer: Per style and reminders:
```suggestion
Expose these headers only if the API is used within the organization. In a public API scenario, this data can be used maliciously to invoke spikes in traffic that leaves your API without any quota left.
```

Reviewer 2: Done! Thanks!
